Лабораторная работа 1. Камаев Сергей

nybc4ik opened the pull request.

nybc4ik committed:
Лабораторная работа 1. Камаев Сергей

nybc4ik committed:
Update data.yaml

Случайно задел слово другого человека, поэтому всё вернул на место!

nybc4ik committed:
Update data.yaml

Попытка пройти тесты путём добавления - и чего-то ещё (?)

nybc4ik reviewed (commented):
Попытка пройти тесты

BaldiSlayer committed:
fix

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
   - 'Какая грамматика называется автоматной?
 
     '
+- answer: ' Накачка языка - это возможность многократного повторения некоторой подстроки
```
@BaldiSlayer , думаю, это вливание сделает жизнь ещё интереснее)

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
     '
+- answer: ' Накачка языка - это возможность многократного повторения некоторой подстроки
+    в любой строке подходящей длины любого бесконечного автоматного языка. Точная
+    формулировка: Для каждой достаточно длинной строки в обычном языке должна существовать
```
Обычный язык - это ordinary в терминологии Охотина?
Это галлюцинация.
Уж если пишете про накачки, то сравнивайте разные типы накачек в языках: синхронные, симметричные, регулярные.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  author: Камаев Сергей
+  id: 169
+  questions:
+  - Что такое накачка языка?
```
Какого языка?

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  - Что такое накачка языка?
+- answer: ' Пусть G — КС-грамматика в форме Хомского. Тогда существует p ∈ N такое,
+    что любое слово w ∈ L(G) длины не меньше p имеет представление вида x1y1zy2x2,
+    где |y1y2| ⩾ 1, |y1zy2| ⩽ p, и все слова вида x1y^k1zy^k2x2 также принадлежат
```
Хотя бы примеры накачек КС-языков могли бы как-то это спасти

TonitaN replied:
Это дубль. И НФХ тут вообще не нужна.

TonitaN replied:
Так и не исправленный дубль

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  author: Камаев Сергей
+  id: 170
+  questions:
+  - Лемма о накачке КС языков
```
Переименовать и упомянуть кванторные игры в ВОПРОСЕ

TonitaN reviewed (changes requested):
Нужно везде сильно расширить тексты вопросов, иначе это вливание вообще полностью сломает БЗ

nybc4ik committed:
Исправление data.yaml

Исправил язык, теперь он реуглярный

TonitaN commented on data/data.yaml:
```diff
@@ -3029,16 +3029,16 @@
-- answer: ' Накачка языка - это возможность многократного повторения некоторой подстроки
-    в любой строке подходящей длины любого бесконечного автоматного языка. Точная
-    формулировка: Для каждой достаточно длинной строки в обычном языке должна существовать
+- answer: ' Накачка реуглярного языка - это возможность многократного повторения некоторой подстроки
```
*регулярного))))))

molivka commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  id: 179
+  questions:
+  - Может ли накачка применяться рекурсивно?
+- answer: 'Для доказательства того, что для данного контекстно-свободного (КС) языка
```
вот у кого я видела это...
ты просто скопировал ответ ллм?)

судя по звёздочкам = выделение жирным
и специфичному обрамлению формул круглыми скобками

nybc4ik replied:
Так у меня даже не залито в ллм ещё @_@

molivka replied:
забавно, что мой комментарий показался "неоднозначным" 
задам вопрос по-другому: ты сам писал текст ответа, строки 3145-3163?)

nybc4ik replied:
Да, я писал сам, если тебе не нравятся мои **жирненькие** слова, я их уберу, когда выкачу следующие исправления.

molivka commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  author: Камаев Сергей
+  id: 178
+  questions:
+  - Что такое лемма Огдена?
```
дубль вопроса 132

nybc4ik replied:
Спасибо за заботу, постараюсь найти ещё один вопросик, который точно не окажется дублем!

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+    формулировка: Для каждой достаточно длинной строки в реуглярном языке должна существовать
+    средняя часть, которую можно повторять (или наращивать) любое количество раз,
+    чтобы получилась строка, которая всё ещё относится к этому языку. Используется
+    для проверки принадлежности слова к определённому языку. '
```
О, я придумала, как спасти этот вопрос. Сделайте вариант "критерий регулярного языка - возможность накачки достаточно длинных слов в нём в рамках любой достаточно длинной подстроки" (не только в префиксе).
У вас близко к этому.

nybc4ik replied:
Большое спасибо за идею! Теперь я смог исправить этот вопрос.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+    в виде “игры с противником” следующим образом. 1. Мы выбираем язык L, желая доказать,
+    что он не контекстно-свободный. 2. Наш “противник” выбирает заранее неизвестное
+    нам n, поэтому мы должны рассчитывать на любое возможное значение. 3. Мы выбираем
+    z и при этом можем использовать n как параметр. 4. Противник разбивает z на uvwxy,
```
пояснить, что такое z, и что нужно рассматривать все возможные разбиения, но i мы можем выбирать сами.

nybc4ik replied:
Добавил все необходимые пояснения.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  author: Камаев Сергей
+  id: 172
+  questions:
+  - Лемма о накачке для регулярных языков
```
Этот вопрос уже не спасти. Это дубль.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  author: Камаев Сергей
+  id: 174
+  questions:
+  - Применение леммы о накачке с помощью двух противников
```
Почти дословный дубль уже разобранного вопроса про игры в накачку кс языка.
Предлагаю заменить на "кванторную сложность множеств (иерархию Клини-Мостовского)" и кванторную сложность свойства накачиваемости

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+    длины не меньше p имеет представление вида x1y1zy2x2, где |y2| ⩾ 1, |x1y1| ⩽ p,
+    |y2| ⩽ p либо y2 накачивается отдельно, |x2| ⩽ p либо x2 накачивается отдельно,
+    и все слова вида x1y^k1zy^k2x2 также принадлежат L(G).
+  author: Камаев Сергей
```
Первая лемма о накачке - это классическая лемма о накачке (чистый дубль).
Вторая лемма о накачке - это лемма о существовании накачки в достаточно длинном префиксе слова, порождённого КС грамматикой. Если вы не понимаете, как её выводить из свойства древесной регулярности деревьев вывода, лучше не беритесь за неё.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  id: 178
+  questions:
+  - Что такое лемма Огдена?
+- answer: Да, понятие накачки может быть применено рекурсивно к некоторым достаточно
```
Пример нужен

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  author: Камаев Сергей
+  id: 177
+  questions:
+  - Что такое нормальная форма Грейбах?
```
На базе вопроса 127, 128, 143, всё прекрасно отвечается. И даже лучше чем у вас, т.к. вы забыли отметить ситуацию с пустым словом
https://gist.github.com/stewkk/8360e931b26f74c1c7396467f3448660

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+
+    1. **Свойства DCFL**: Детерминированные КС языки (DCFL) замкнуты относительно
+    операций дополнения и пересечения с регулярными языками. Если можно показать,
+    что данный язык не обладает одним из этих свойств, это докажет его недетерминированность.
```
Это галлюцинация. Какая ещё не замкнутость отдельного языка относительно множественных операций?

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  questions:
+  - Может ли накачка применяться рекурсивно?
+- answer: 'Для доказательства того, что для данного контекстно-свободного (КС) языка
+    нельзя построить дпда (DPDA), можно использовать несколько подходов.
```
дпда никто не пишет. МП-автомат

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+    ( a^n b^m c^m ), автомат должен проверить, что ( m = k ). - Однако DPDA не может
+    одновременно хранить в стеке информацию о двух независимых параметрах ( n ) и
+    ( k ) и принимать решение в детерминированной форме. Следовательно, DPDA не может
+    обработать ( L ), и данный язык недетерминирован.
```
Это не доказательство

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,139 @@
+  author: Камаев Сергей
+  id: 180
+  questions:
+  - Как доказать, что для данного кс языка нельзя построить детерминированный pda?
```
Тут всё настолько ужасно, что проще заменить вопрос.

nybc4ik replied:
Заменил вопрос

nybc4ik committed:
Update data.yaml

nybc4ik committed:
Добавил вопрос про длину накачки

BaldiSlayer commented on data/data.yaml:
```diff
@@ -3165,3 +3165,66 @@
   questions:
   - Как доказать, что для данного кс языка нельзя построить детерминированный pda?
+  - answer: '
+    Длина накачки — это число, связанное с леммой о накачке для регулярных языков. 
```
Только для регулярных?

nybc4ik replied:
Спасибо за внимательность, сейчас добавлю и для кс - языков

TonitaN replied:
Ахаха, может, лучше сразу в общем виде? Ещё же минимум для DCFL есть, и линейных КС, и это всё - длины накачки, которые можно охарактеризовать единообразно)

nybc4ik replied:
Привёл определение к общему виду! Пришлось удалить примеры :(

nybc4ik committed:
Привёл к общему виду

TonitaN commented on data/data.yaml:
```diff
@@ -3166,64 +3166,9 @@
-    Таким образом во всех возможных разбиениях строки s = a^pb^p условие леммы о накачке нарушается. Следовательно, язык не является регулярным.
-
-
+    Этот параметр позволяет доказать принадлежность языка к определенному классу.
```
Принадлежность?)

TonitaN commented on data/data.yaml:
```diff
@@ -3166,64 +3166,9 @@
   - answer: '
-    Длина накачки — это число, связанное с леммой о накачке для регулярных языков. 
+    Длина накачки — это число, связанное с леммой о накачке для регулярных или контекстно-свободных языков. 
     Она обозначается как \(p\) и указывает минимальную длину строки, при которой можно применить лемму о накачке. 
```
И всё-таки у всех лемм о накачке есть кое-что общее) Вот хотелось бы увидеть, что.

nybc4ik replied:
Я добавил общие детали для всех лемм о накачке

TonitaN replied:
Кстати, обозначение p здесь не существенно, потому что оно дальше нигде не используется.

nybc4ik committed:
Исправление ошибок

nybc4ik committed:
Update data.yaml

nybc4ik committed:
Исправил вопрос 

Упомянул в вопросе критерий регулярности языка

nybc4ik committed:
Исправил ещё один вопрос про игру противников

Пояснил, что такое z и то, что мы можем выбирать i самостоятельно

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,151 @@
+- answer: 'Критерий регулярного языка заключается в том, что в любом достаточно длинном слове этого языка можно найти подстроку, 
+    которую можно накачивать любое количество раз, и полученные слова всё ещё будут принадлежать этому языку. 
+    Это условие распространяется не только на префиксы, но и на любую достаточно длинную подстроку.
+    Точная формулировка: Для каждой достаточно длинной строки в регулярном языке существует такая её внутренняя часть, 
```
Если просто существует какая-то накачиваемая внутренняя часть любой достаточно длинной строки, то это более слабое утверждение, чем классическая лемма о накачке. Забыли кое-что тут

nybc4ik replied:
Добавил информации касательно подстроки.

TonitaN replied:
И получилась стандартная лемма о накачке, а не критерий регулярности.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,151 @@
+  author: Камаев Сергей
+  id: 169
+  questions:
+  - Что такое критерий регулярности языка?
```
Их много, уточнить

nybc4ik replied:
Уточнил этот вопрос, также добавил ещё один вопрос, но с другим критерием.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,151 @@
+  - Лемма о накачке КС языков
+- answer: Лемма о накачке для КС-языков, как и для регулярных языков, используется
+    в виде “игры с противником” следующим образом. 1. Мы выбираем язык L, желая доказать,
+    что он не контекстно-свободный. 2. Наш “противник” выбирает заранее неизвестное
```
Именно "не контекстно-свободный"?

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,151 @@
+    в виде “игры с противником” следующим образом. 1. Мы выбираем язык L, желая доказать,
+    что он не контекстно-свободный. 2. Наш “противник” выбирает заранее неизвестное
+    нам n, поэтому мы должны рассчитывать на любое возможное значение. 3. Мы выбираем слово 
+    z, которое принадлежит языку, оно должно быть не меньше n, при этом z зависит от n, 
```
что значит "слово не меньше n"?

TonitaN replied:
X__X

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,151 @@
+  id: 170
+  questions:
+  - Лемма о накачке КС языков
+- answer: Лемма о накачке для КС-языков, как и для регулярных языков, используется
```
Нужно аккуратнее это сформулировать. Можно унифицировать это описание для обеих лемм о накачке, но тогда нужно определять действия по-другому, более общо.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,151 @@
+    так что мы можем использовать n как параметр (лучше всего выбирать z таким образом, чтобы, накачивая или наоборот укорачивая, получить противоречие, ведь в этом наша цель).
+    4. Противник разбивает z на uvwxy, соблюдая ограничения |vwx| ≤ n и vx ≠ ε. 
+    Противник может может выбрать любое разбиение, поэтому очень важно учесть их все при выборе z.
+    5. Мы “выигрываем”, если можем, самостоятельно выбрать нужное нам значение i и затем показать, что uv^iwx^i y не принадлежит языку L.
```
Вспомнить про кванторы, и за какой квантор и в какой формуле мы "играем"

TonitaN reviewed (changes requested):
+ ещё старые замечания никуда не делись

nybc4ik committed:
Исправил вопросы

Удалил плохой вопрос, добавил информации в хороший, продолжаю исправления :(

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,134 @@
+  id: 179
+  questions:
+  - Может ли накачка применяться рекурсивно?
+- answer: 'Язык является регулярным, если число классов эквивалентности конечно.'
```
Относительно какого отношения? ОМГ

TonitaN reviewed (commented):
Почему все думают, что если замечания просто проигнорировать, они сами собой исправятся и зачтутся?
Это работает не так.
Лимит запросов к МАТ исчерпан.

nybc4ik committed:
Исправил критерий регулярности

nybc4ik committed:
Исправление вопросов

Исправил вопросы про накачку КС языков

TonitaN commented:
По 10 раз не буду тут повторять уже 100 раз повторённое.

Готовьте вариант для окончательной сдачи, перепроверок уже не будет, только одна. И готовьте с учётом всего, о чём шла речь выше и в общем относительно заполнения БЗ.

nybc4ik committed:
Правки вопросов и замена дублей

Внёс все необходимые правки, заменил дубли и добавил новые вопросы.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+  author: Камаев Сергей
+  id: 170
+  questions:
+  - 'Лемма о накачке КС языков'
```
Дубль id 50 , о чём уже говорилось.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+  id: 172
+  questions:
+  - 'Является ли язык правильной скобочной последовательности контекстно-свободным? Если да, то приведи пример грамматики, задающей этот язык. Если нет, то обоснуй почему?'
+- answer: 'Ограниченный квантор - это такой квантор, в котором переменная квантора ограничивается термом, при этом для ограничения используется < или <=. 
```
Тут формулировку лучше уточнить: подкванторная переменная  может пробегать только конечное множество значений.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+  questions:
+  - 'Что такое кванторная сложность множеств и иерархия Клини-Мостовского?'
+- answer: 'Кванторная структура свойства накачиваемости может быть представлена следующим образом:
+    Критерий регулярности языка: ∃p ∀w ∈ L |w| ≥ p ∃u, x, y, z: w = uxyz и |u| ≤ |w| - p и |xy| ≤ p и |y| > 0 ∀k ≥ 0 uxyᵏz ∈ L
```
Лучше переписать со скобочками в предварённой нормальной форме, а не словами.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+- answer: 'Кванторная структура свойства накачиваемости может быть представлена следующим образом:
+    Критерий регулярности языка: ∃p ∀w ∈ L |w| ≥ p ∃u, x, y, z: w = uxyz и |u| ≤ |w| - p и |xy| ≤ p и |y| > 0 ∀k ≥ 0 uxyᵏz ∈ L
+    В критерии для регулярных языков есть три неограниченных квантора, которые дают бесконечные альтернативы ∃p ∈ ℕ  ∀w |w| > p (здесь далее идет ограничивающий квантор на разбиение w на подслова, потому что разбиений слова — конечное число) ∀k k ⩾ 0 (...)
+    Но два повторяющихся неограниченных квантора ∀ засчитываются как один, так как нет перехода от одного типа квантора к другому. Поэтому класс все равно остается Σ₂ — второй класс, даже несмотря на то, что неограниченных кванторов три.
```
Как один над парой значений - точнее.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+S -> S1 | S2
+В этом правиле S1 порождает все строки из L1, а S2 из L2. Все остальные правила из G1 и G2 добавляем без изменений. (Но если в грамматиках будут одинаковые нетерминалы, их нужно переименовать!)  
+Таким образом новая грамматика порождает все строки из L1 и L2.
+Вывод- объединение контекстно-свободных языков всегда контекстно-свободно'
```
Дубль id: 165

TonitaN replied:
Всё ещё :(

TonitaN replied:
Было же

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+  id: 177
+  questions:
+  - 'Приведи определение линейного языка и его пример.'
+- answer:  'Нет, не всегда. Пусть есть контекстно-свободные языки L1 = a^nb^nc^k и L2 = a^kb^nc^n (n,k => 0). 
```
Чисто формально тут написаны определения языков, содержащих единственное слово. Нужно в свёрточной форме сделать.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+  id: 176
+  questions:
+  - 'Всегда ли объединение двух контекстно-свободных языков является контекстно-свободным языком?'
+- answer: 'Линейный язык — это язык, который может быть порожден линейной грамматикой. Линейная грамматика — это контекстно-свободная грамматика, 
```
Строго говоря, это дубль id: 246, но его можно спасти, переделав в лемму о накачке для линейных кс-языков.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+  questions:
+  -  'Является ли объединение детерминированных контекстно-свободных языков детерминированным или недетерминированным?'
+  - answer: '
+    'Язык \(L\) является регулярным тогда и только тогда, когда отношение эквивалентности по продолжениям, определённое так:  
```
Дубль очень многих вопросов.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+    author: Камаев Сергей
+  id: 180
+  questions: 'Как формулируется критерий регулярности языка в терминах классов эквивалентности?'
+    - answer: ' Длина накачки — это число, связанное с леммой о накачке. 
```
*с леммой о накачке для некоторого класса формальных языков

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+    K -> bKc | ε 
+Теперь если сделать пересечение этих двух языков, можно получить язык L3= a^nb^nc^n, который не является контекстно-свободным, что нетрудно проверить используя лемму о накачке для КС языков. 
+Таким образом пересечение двух контекстно-свободных языков не всегда является контекстно-свободным.'
+  author: Камаев Сергей
```
Частичный дубль id: 165.
Но его можно кое-как спасти, добавив вопрос "доказательство незамкнутости КС-языков относительно пересечения".

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,154 @@
+  author: Камаев Сергей
+  id: 179
+  questions:
+  -  'Является ли объединение детерминированных контекстно-свободных языков детерминированным или недетерминированным?'
```
Дубль id: 222. Подумайте, как изменить.

TonitaN reviewed (changes requested):
Всё ещё много дублей и есть погрешности.

Но хотя бы видно, что были попытки исправиться, и местами даже удачные. Поэтому даю вам шанс доработать имеющиеся и разобраться с дублями сегодня до 21.

nybc4ik committed:
Исправления ошибок

Прошу прощения за задержку с исправлениями! Постарался исправить ошибки, убрать дубли и придумать новые вопросы.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,122 @@
+  questions:
+  - 'Что такое кванторная сложность множеств и иерархия Клини-Мостовского?'
+- answer: 'Кванторная структура свойства накачиваемости может быть представлена следующим образом:
+    Критерий регулярности языка: ∃p ∀w ∃u∃x∃y∃z ∀k¬(p ∈ ℕ) ⋁ ¬(w ∈ L) ⋁ ¬(|w| ≥ p) ⋁ ¬(w = uxyz) ⋁ ¬(|u| ≤ |w| - p) ⋁ (uxyᵏz ∈ L)
```
Самое главное забыли, в итоге формула стала уже не в ПНФ.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,122 @@
+  questions:
+  - 'Доказательство незамкнутости КС-языков относительно пересечения'
+- answer: 'Язык является линейным тогда и только тогда, когда найдётся МП-автомат, который допускает язык по пустому стеку,
+при этом в процессе работы автомата для строчки из языка стек сперва растёт, затем уменьшается, и на этом работа автомата заканчивается.'
```
Непонятно, что значит "сначала растёт, а потом уменьшается". Нужно более точно.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,122 @@
+- answer: 'Линейный язык — это язык, который может быть порожден линейной грамматикой. Линейная грамматика — это контекстно-свободная грамматика, 
+  в которой в каждом правиле вывода не более одного нетерминала в правой части. 
+  Лемма о накачке линейных КС языков: 
+  Если L линейный КС язык, тогда существует такая константа n, что если z — произвольная цепочка из L, длина которой не
```
Тут ошибка в формулировке. В частности, я не могу накачать `a^n b^n` при таком выборе разбиения, очевидно линейный язык.

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,122 @@
+  id: 169
+  questions:
+  - 'Какой критерий используется для определения регулярности языка в контексте леммы о накачке?'
+- answer: 'Временная сложность алгоритма удаления бесполезных символов в контекстно-свободной грамматике определяется двумя алгоритмами, которые работают один за одним и запускаются один раз. 
```
Почему именно сначала непорождающих, потом недостижимых? Пояснить.
Что за специальные структуры данных?
Что такое `n`?

Так много вопросов, так мало ответов...

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,122 @@
+  author: Камаев Сергей
+  id: 178
+  questions:
+  - 'Доказательство незамкнутости КС-языков относительно пересечения'
```
Добавьте здесь ещё вопрос, который был: `пересечение КС-языков`

TonitaN reviewed (commented):
Лимит запросов к MAT всё.
И так у вас был повышенный сегодня)

nybc4ik committed:
Update data.yaml

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
   - 'Какая грамматика называется автоматной?
 
     '
+- answer: 'Язык L является регулярным тогда и только тогда, когда существует число m такое, что для любого слова w из этого языка длины m выполняется следующее: для любого его разбиения на префикс и суффикс, 
```
Неточно. Где именно внутри суффикса найдётся такое слово?

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  author: Камаев Сергей
+  id: 172
+  questions:
+  - 'Является ли язык правильной скобочной последовательности контекстно-свободным? Если да, то приведи пример грамматики, задающей этот язык. Если нет, то обоснуй почему?'
```
Давайте уже без языков Дика, про них уже не один элемент БЗ есть, выражающий их в каких только можно формализмах. И да, что они КС, тоже есть в ответах БЗ

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  id: 172
+  questions:
+  - 'Является ли язык правильной скобочной последовательности контекстно-свободным? Если да, то приведи пример грамматики, задающей этот язык. Если нет, то обоснуй почему?'
+- answer: 'Ограниченный квантор - это такой квантор, в котором подкванторная переменная ограничена некоторыми логическими условиями, таким образом, что она пробегает только конечное множество значений.  
```
Вот здесь добавить, чем они обозначаются дальше

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  Иерархия Клини-Мостовского описывает уровни сложности таких представлений, то есть классифицирует предикаты и множества натуральных чисел на основе их кванторной сложности. Она состоит из уровней Σₙ и Πₙ для n ≥ 0:
+  - Σₙ: класс предикатов, которые можно выразить в форме ∃x₁ ∀x₂ ∃x₃ ... Qxₙ φ(x₁, x₂, ..., xₙ), где Q — квантор существования или всеобщности, а φ — предикат из класса Δ₀.
+  - Πₙ: класс предикатов, которые можно выразить в форме ∀x₁ ∃x₂ ∀x₃ ... Qxₙ φ(x₁, x₂, ..., xₙ), где Q — квантор существования или всеобщности, а φ — предикат из класса Δ₀.
+  Например, предикат, выраженный формулой вида ∃x₁ ∀x₂ ∃x₃ φ(x₁, x₂, x₃), принадлежит классу Σ₃, поскольку начинается с трех чередующихся кванторов, начиная с ∃.'
```
Лучше: "содержит три чередующихся группы кванторов, начиная с квантора существования". Т.к. однородные кванторы мы группируем

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  - 'Является ли язык правильной скобочной последовательности контекстно-свободным? Если да, то приведи пример грамматики, задающей этот язык. Если нет, то обоснуй почему?'
+- answer: 'Ограниченный квантор - это такой квантор, в котором подкванторная переменная ограничена некоторыми логическими условиями, таким образом, что она пробегает только конечное множество значений.  
+  Неограниченные кванторы - это все остальные кванторы, которые не являются ограниченными.
+  Кванторная сложность множества определяется количеством и чередованием неограниченных кванторов существования (∃) и всеобщности (∀) в формуле, описывающей это множество. Чем больше чередований этих кванторов, тем выше кванторная сложность. 
```
Только числом чередований она определяется, то есть количество не важно

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  questions:
+  - 'Что такое кванторная сложность множеств и иерархия Клини-Мостовского?'
+- answer: 'Кванторная структура свойства накачиваемости может быть представлена следующим образом:
+    Критерий регулярности языка: ∃p ∀w ∃(u, x, y, z) ∀k (p ∈ ℕ ∧ ((w ∉ L) ∨ (w = uxyz ∧ |u| ≤ |w| - p ∧ ((k ∉ ℕ ∪ {0}) ∨ uxyᵏz ∈ L) ) ) ) 
```
Если уж всё формально расписывать, то потеряли как минимум ограничение, что y непуст, и то самое ограничение на x, про которое я уже писала

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  author: Камаев Сергей
+  id: 174
+  questions:
+  - 'Какова кванторная сложность свойства накачиваемости?'
```
Свойства накачиваемости какого из всех?

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  questions:
+  - 'Какова кванторная сложность свойства накачиваемости?'
+- answer: 'Нет, не всегда существует ДКА с тем же числом состояний, который распознаёт тот же язык, что и данный НКА.
+Например язык (1|0)*1(1|0) при построении НКА имеет 3 состояния, при этом для ДКА ему же потребуется минимум 4 состояния. Это связано с тем, что НКА позволяет с помощью недетерменима совершать сразу несколько переходов 
```
1. Недетерменизм - друг фунтированости
2. совершать сразу что?
3. языки с меньшим количеством состояний - человек поймёт, а LLM свяжет дополнение с ближайшим определяемым словом, то есть это не автоматы уже будут с меньшим количеством состояний, а именно языки
Пожалуйста, вычитывайте)

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  id: 177
+  questions:
+  - 'Лемма о накачке линейных КС языков'
+- answer:  'Доказательство: Пусть есть контекстно-свободные языки L1 = {a^nb^nc^k | n,k => 1} и L2 = {a^kb^nc^n | n,k => 1}. 
```
=> - это в народе следование, а не "больше или равно". Безопаснее сделать `>0`

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  questions:
+  - 'Доказательство незамкнутости КС-языков относительно пересечения'
+- answer: 'Язык является линейным тогда и только тогда, когда найдётся МП-автомат, допускающий его по пустому стеку, такой, что для любого слова из этого языка:  
+1) В процессе чтения высота стека увеличивается с помощью операции push, пока не достигнет некоторой высоты  
```
а давайте мы здесь по-русски скажем про изменение стека? Или хотя бы дополнительно расшифруем?

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+
+Язык L является линейным тогда и только тогда, когда существует МП-автомат M, допускающий L по пустому стеку, и существует целочисленная константа c ≥ 0 такая, 
+что для любого слова w ∈ L существует вычисление автомата M, принимающее w (приводящее к пустому стеку после прочтения w), 
+в котором высота стека в любой момент этого вычисления не превышает c.
```
А это баг. Ну смотрите, чем я могу ограничить `c` в классическом языке слов `a^n b^n`? Зато знак неравенства здесь красивый)

TonitaN commented on data/data.yaml:
```diff
@@ -3031,3 +3029,136 @@
+  questions:
+  -  'Критерий линейности языка?'
+    - answer: 'Длина накачки — это число, связанное с леммой о накачке для некоторого класса формальных языков. 
+    Она обозначает минимальную длину строки из языка, при которой можно применить преобразование из леммы о накачке и опять получить строку из языка.'
```
Какое именно преобразование? Оно типично для всех лемм, которые носят название "о накачке", потому что напрямую следует из этого названия

TonitaN reviewed (changes requested):
There we go again...
